pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -140,3 +137,31 @@ func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *trainer.TrainJob)
 
 	return nil
 }
+
+func (t *Torch) Validate(runtimeJobTemplate client.Object, runtimeInfo *runtime.Info, oldObj, newObj *trainer.TrainJob) (admission.Warnings, field.ErrorList) {
+	var allErrs field.ErrorList
+	specPath := field.NewPath("spec")
+
+	if newObj.Spec.Trainer != nil {
+		numProcPerNodePath := specPath.Child("trainer").Child("numProcPerNode")
+		if runtimeInfo.RuntimePolicy.MLPolicy != nil &&
+			runtimeInfo.RuntimePolicy.MLPolicy.Torch != nil && newObj.Spec.Trainer.NumProcPerNode != nil {
+			numProcPerNode := *newObj.Spec.Trainer.NumProcPerNode
+			if numProcPerNode.Type == intstr.String {
+				allowedStringValList := []string{"auto", "cpu", "gpu"}
```

Comment:
could we use CEL validation instead of this webhook?

Reply:
same comment as above

Reply:
Oh, that's right. Thank you for clarifying that!